src/utils/test/clm_time_manager_test/test_clm_time_manager.pf

```diff
@@ -577,4 +577,35 @@ contains
 
   end subroutine bad_hilontolocal_time
 
+  @Test
+  subroutine check_is_doy_in_interval(this)
+    class(TestTimeManager), intent(inout) :: this
+
+    integer :: start, end
+
+    start = 100
+    end = 300
+    @assertTrue(is_doy_in_interval(start, end, start))
+    @assertTrue(is_doy_in_interval(start, end, end))
+    @assertTrue(is_doy_in_interval(start, end, 200))
+    @assertFalse(is_doy_in_interval(start, end, 35))
+    @assertFalse(is_doy_in_interval(start, end, 350))
+
+    start = 300
+    end = 100
+    @assertTrue(is_doy_in_interval(start, end, start))
+    @assertTrue(is_doy_in_interval(start, end, end))
+    @assertFalse(is_doy_in_interval(start, end, 200))
+    @assertTrue(is_doy_in_interval(start, end, 35))
+    @assertTrue(is_doy_in_interval(start, end, 350))
+
+    start = 300
+    end = 300
+    @assertTrue(is_doy_in_interval(start, end, start))
+    @assertTrue(is_doy_in_interval(start, end, end))
+    @assertFalse(is_doy_in_interval(start, end, 200))
+    @assertFalse(is_doy_in_interval(start, end, 350))
```

Comment:
This is a great set of unit tests of this function – I'm glad you added all of these to cover a variety of conditions.

I don't know whether to make this next comment, but I'll go ahead and make it and then walk it back a little afterwards: I tend to subscribe to the (common, I think) unit testing guidance of having each test just test a single scenario. The advantages of this are: (1) if you get a failure, the test is (hopefully) clearly named so that you can see based on the failure exactly what condition failed; and (2) all scenarios are run, in contrast to a situation like this one where the first failed assertion will cause the test to bail, so you have a less clear picture of what works and what doesn't. That said, I have come to feel less strongly about this over time, and in this particular case the (small but still non-zero) overhead of splitting each test into its own subroutine may not be worth it. So my overall feeling is: we should strive to follow the one-scenario-per-test guidance in general, but there are always times when it's worth breaking a rule, and this might be one of those times.

Reply:
I definitely see what you're saying, but since you've provided me an out from having to split this into 15 separate tests (plus 6 more for `check_is_today_in_doy_interval()`), I'm taking it! (I'm sure I could reduce the number of tests without compromising coverage, but I'd rather err on the side of more tests than necessary.)

However, I did split `check_is_doy_in_interval()` into three separate tests based on the relative values of `start` and `end` (eded771). Hopefully that's a good compromise.